lib/ansible/module_utils/vultr.py

```diff
@@ -0,0 +1,242 @@
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import time
+import urllib
+from ansible.module_utils.six.moves import configparser
+from ansible.module_utils._text import to_text
+from ansible.module_utils.urls import fetch_url
+
+
+VULTR_API_ENDPOINT = "https://api.vultr.com"
+
+
+def vultr_argument_spec():
+    return dict(
+        api_key=dict(default=os.environ.get('VULTR_API_KEY'), no_log=True),
+        api_timeout=dict(type='int', default=os.environ.get('VULTR_API_TIMEOUT') or 60),
+        api_retries=dict(type='int', default=os.environ.get('VULTR_API_RETRIES') or 5),
+        api_account=dict(default=os.environ.get('VULTR_API_ACCOUNT') or 'default'),
+        validate_certs=dict(default=True, type='bool'),
+    )
+
+
+class Vultr(object):
+
+    def __init__(self, module, namespace):
+        self.module = module
+
+        # Namespace use for returns
+        self.namespace = namespace
+        self.result = {
+            'changed': False,
+            namespace: dict(),
+            'diff': dict(before=dict(), after=dict())
+        }
+
+        # For caching HTTP API responses
+        self.api_cache = dict()
+
+        # Reads the config from vultr.ini
+        try:
+            config = self.read_ini_config()
+        except KeyError:
+            config = {}
+
+        self.api_config = {
+            'api_key': self.module.params.get('api_key') or config.get('key'),
+            'api_timeout': self.module.params.get('api_timeout') or config.get('timeout'),
+            'api_retries': self.module.params.get('api_retries') or config.get('retries'),
+        }
+
+        # Common vultr returns
+        self.result['vultr_api'] = {
+            'api_account': self.module.params.get('api_account'),
+            'api_timeout': self.api_config['api_timeout'],
+        }
+
+        # Headers to be passed to the API
+        self.headers = {
+            'API-Key': "%s" % self.api_config['api_key'],
+            'User-Agent': "Ansible Vultr",
+            'Accept': 'application/json',
+        }
+
+    def read_ini_config(self):
+        ini_group = self.module.params.get('api_account')
+
+        keys = ['key', 'timeout', 'retries']
+        env_conf = {}
+        for key in keys:
+            if 'VULTR_API_%s' % key.upper() not in os.environ:
+                break
+            else:
+                env_conf[key] = os.environ['VULTR_API_%s' % key.upper()]
+        else:
+            return env_conf
+
+        paths = (
+            os.path.join(os.path.expanduser('~'), '.vultr.ini'),
+            os.path.join(os.getcwd(), 'vultr.ini'),
+        )
+        if 'VULTR_API_CONFIG' in os.environ:
+            paths += (os.path.expanduser(os.environ['VULTR_API_CONFIG']),)
+        if not any([os.path.exists(c) for c in paths]):
```

Comment:
note, this is not a blocker but you can use a generator expression here instead of a list comprehension.  it could save on some system calls (calling os.path.exists) but I doubt that will be a noticable optimization so it's really just style.